datadog_lambda/trigger.py

```diff
@@ -0,0 +1,205 @@
+        if event_record.get("cf"):
+            event_source = "cloudfront"
+
+    if event_source in EVENT_SOURCES:
+        event_source = event_source.replace("aws:", "")
+    return event_source
+
+
+def parse_event_source_arn(source, event, context):
+    """
+    Parses the trigger event for an available ARN. If an ARN field is not provided
+    in the event we stitch it together.
+    """
+    split_function_arn = context.invoked_function_arn.split(":")
+    region = split_function_arn[3]
+    account_id = split_function_arn[4]
+    aws_arn = get_arn_region_identifier(region)
+
+    event_record = get_first_record(event)
+    # e.g. arn:aws:s3:::lambda-xyz123-abc890
+    if source == "s3":
+        return event_record.get("s3")["bucket"]["arn"]
+
+    # e.g. arn:aws:sns:us-east-1:123456789012:sns-lambda
+    if source == "sns":
+        return event_record.get("Sns")["TopicArn"]
+
+    # e.g. arn:aws:cloudfront::123456789012:distribution/ABC123XYZ
+    if source == "cloudfront":
+        distribution_id = event_record.get("cf")["config"]["distributionId"]
+        return "arn:{}:cloudfront::{}:distribution/{}".format(
+            aws_arn, account_id, distribution_id
+        )
+
+    # e.g. arn:aws:apigateway:us-east-1::/restapis/xyz123/stages/default
+    if source == "api-gateway":
+        request_context = event.get("requestContext")
+        return "arn:{}:apigateway:{}::/restapis/{}/stages/{}".format(
+            aws_arn, region, request_context["apiId"], request_context["stage"]
+        )
+
+    # e.g. arn:aws:elasticloadbalancing:us-east-1:123456789012:targetgroup/lambda-xyz/123
+    if source == "application-load-balancer":
+        request_context = event.get("requestContext")
+        return request_context.get("elb")["targetGroupArn"]
+
+    # e.g. arn:aws:logs:us-west-1:123456789012:log-group:/my-log-group-xyz
+    if source == "cloudwatch-logs":
+        with gzip.GzipFile(
+            fileobj=BytesIO(base64.b64decode(event["awslogs"]["data"]))
+        ) as decompress_stream:
+            data = b"".join(BufferedReader(decompress_stream))
+        logs = json.loads(data)
+        log_group = logs.get("logGroup", "cloudwatch")
+        return "arn:{}:logs:{}:{}:log-group:{}".format(
+            aws_arn, region, account_id, log_group
+        )
+
+    # e.g. arn:aws:events:us-east-1:123456789012:rule/my-schedule
+    if source == "cloudwatch-events" and event.get("resources"):
+        return event.get("resources")[0]
+
+
+def get_event_source_arn(source, event, context):
+    event_source_arn = event.get("eventSourceARN") or event.get("eventSourceArn")
+
+    event_record = get_first_record(event)
+    if event_record:
+        event_source_arn = event_record.get("eventSourceARN") or event_record.get(
+            "eventSourceArn"
+        )
+
+    if event_source_arn is None:
+        event_source_arn = parse_event_source_arn(source, event, context)
+
+    return event_source_arn
+
+
+def get_http_tags(event):
```

Comment:
nits: `extract_http_tags`? One way to determine a good function name is to reduce it from the docstring, in this case `Extracts HTTP facet tags from the triggering event`.